### ISSUE TITLE
an execution of pipeline via master api is not working

### ISSUE THREAD

**Issue by Jan-Marcek** [UnifiedViews/Core#558]
Hello,

version: 2.3.0  
UPDATE: wrong commit id: https://github.com/UnifiedViews/Core/commit/2332da93636b74b32b6225b5b421c1f91e4ee6da see a comment

It is related to open data node. Please, try if an execution via rest api is still working.

Thanks a lot.

```
2015-12-11 10:15:16,825 [http-bio-127.0.0.1-28080-exec-1] WARN  org.eclipse.persistence.logging.default - 2015-12-11 10:15:16.825--UnitOfWork(2048110751)--
2015-12-11 10:15:16,826 [http-bio-127.0.0.1-28080-exec-1] ERROR eu.unifiedviews.master.model.ApiException - API exception raised: status: Internal Server Error, errorMessage: Pipeline execution cannot be created., technicalMessage: During synchronization a new object was found through a relationship that was not marked cascade PERSIST: cz.cuni.mff.xrg.odcs.commons.app.pipeline.PipelineExecution@2eaf0b78.; nested exception is java.lang.IllegalStateException: During synchronization a new object was found through a relationship that was not marked cascade PERSIST: cz.cuni.mff.xrg.odcs.commons.app.pipeline.PipelineExecution@2eaf0b78.
eu.unifiedviews.master.model.ApiException: HTTP 500 Internal Server Error
        at eu.unifiedviews.master.api.ExecutionResource.createPipelineExecution(ExecutionResource.java:209) [ExecutionResource.class:na]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_91]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_91]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_91]
        at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_91]
        at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:151) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:171) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:195) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:104) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:384) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:342) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:101) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:271) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271) [jersey-common-2.13.jar:na]
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267) [jersey-common-2.13.jar:na]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:315) [jersey-common-2.13.jar:na]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:297) [jersey-common-2.13.jar:na]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:267) [jersey-common-2.13.jar:na]
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:297) [jersey-common-2.13.jar:na]
        at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:254) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1030) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:373) [jersey-container-servlet-core-2.13.jar:na]
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:381) [jersey-container-servlet-core-2.13.jar:na]
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:344) [jersey-container-servlet-core-2.13.jar:na]
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:221) [jersey-container-servlet-core-2.13.jar:na]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:305) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:225) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:123) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:472) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:168) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:98) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:927) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:118) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:407) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1002) [tomcat-coyote-7.0.28.jar:7.0.28]
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:579) [tomcat-coyote-7.0.28.jar:7.0.28]
        at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:310) [tomcat-coyote-7.0.28.jar:7.0.28]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_91]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_91]
        at java.lang.Thread.run(Thread.java:745) [na:1.7.0_91]

```

**Comment by Jan-Marcek** [comment]
UPDATE:
The right commit is https://github.com/UnifiedViews/Core/commit/e44d1623f649aef3875bcbd4f9d497b759a0d178. The one in issue is wrong.

**Comment by tomas-knap** [comment]
@Jan-Marcek  Please add also steps to reproduce - you created pipeline X and then tried to execute it?

**Comment by Jan-Marcek** [comment]
@tomas-knap . It is just a regular call of rest api from a ckan extension.
You can see it on the https://github.com/OpenDataNode/ckanext-odn-pipeline/blob/master/ckanext/pipeline/uv_helper.py#L192-L200

In fact, this is what is called. If you run this you will see the same result in master.log
```
curl  -X POST http://127.0.0.1:28080/master/api/1/pipelines/8/executions -d '{"debugging": "false", "userActorExternalId": "casadmin", "userExternalId": "comsode"}'  -H "Content-Type: application/json" -u master:commander
```

You can see from the previous comment that the problem is in the method ```createPipelineExecution```: more precisely : ``` During synchronization a new object was found through a relationship that was not marked cascade PERSIST: ``` -> this is a common message of JPA or Hibernate.

Have you tried an execution of pipeline via rest api after db migration to JPA/Hibernate solution? Does work it on your instances?

**Comment by tomas-knap** [comment]
I am testing it on UV installation, having one pipeline created. The executing:  

```
curl  -X POST http://127.0.0.1:8080/master/api/1/pipelines/1/executions -d '{"debugging": "false",  "userExternalId": "http://www.johnadmin.cz"}'  -H "Content-Type: application/json" -u admin:commander
{"id":3,"status":"QUEUED","orderNumber":1,"start":null,"end":null,"schedule":null,"stop":false,"lastChange":"2015-12-16T11:42:19.484+0100","userExternalId":"http://www.johnadmin.cz","userActorExternalId":null,"debugging":false}
```

which works correctly

**Comment by Jan-Marcek** [comment]
I got same error

version: https://github.com/UnifiedViews/Core/commit/d4e8971b0a7b73692314f56d3af3f2559e4ca66b

```
2015-12-16 16:31:20,806 [http-bio-127.0.0.1-28080-exec-7] ERROR eu.unifiedviews.master.model.ApiException - API exception raised: status: Internal Server Error, errorMessage: Pipeline execution cannot be created., technicalMessage: During synchronization a new object was found through a relationship that was not marked cascade PERSIST: cz.cuni.mff.xrg.odcs.commons.app.pipeline.PipelineExecution@64618435.; nested exception is java.lang.IllegalStateException: During synchronization a new object was found through a relationship that was not marked cascade PERSIST: cz.cuni.mff.xrg.odcs.commons.app.pipeline.PipelineExecution@64618435.
eu.unifiedviews.master.model.ApiException: HTTP 500 Internal Server Error
        at eu.unifiedviews.master.api.ExecutionResource.createPipelineExecution(ExecutionResource.java:209) [ExecutionResource.class:na]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_91]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_91]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_91]
        at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_91]
        at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:151) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:171) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:195) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:104) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:384) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:342) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:101) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:271) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271) [jersey-common-2.13.jar:na]
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267) [jersey-common-2.13.jar:na]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:315) [jersey-common-2.13.jar:na]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:297) [jersey-common-2.13.jar:na]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:267) [jersey-common-2.13.jar:na]
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:297) [jersey-common-2.13.jar:na]
        at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:254) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1030) [jersey-server-2.13.jar:na]
        at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:373) [jersey-container-servlet-core-2.13.jar:na]
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:381) [jersey-container-servlet-core-2.13.jar:na]
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:344) [jersey-container-servlet-core-2.13.jar:na]
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:221) [jersey-container-servlet-core-2.13.jar:na]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:305) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:225) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:123) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:472) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:168) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:98) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:927) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:118) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:407) [tomcat-catalina-7.0.28.jar:7.0.28]
        at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1002) [tomcat-coyote-7.0.28.jar:7.0.28]
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:579) [tomcat-coyote-7.0.28.jar:7.0.28]
        at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:310) [tomcat-coyote-7.0.28.jar:7.0.28]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_91]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_91]
        at java.lang.Thread.run(Thread.java:745) [na:1.7.0_91]
2015-12-16 16:31:21,170 [http-bio-127.0.0.1-28080-exec-9] ERROR eu.unifiedviews.master.model.ApiException - API exception raised: status: Not Found, errorMessage: Pipeline execution not found for pipeline!, technicalMessage: null pipeline executions for pipeline with ID=8!
```

**Comment by tomas-knap** [comment]
Strange, this must be solved before ODN working with UV 2.3.+ is released
